cli/src/main/java/com/devonfw/tools/ide/io/FileAccess.java

```diff
@@ -51,6 +51,32 @@ public interface FileAccess {
    */
   void move(Path source, Path targetDir);
 
+  /**
+   * Symbolic links can point to relative or absolute paths. Here the link is converted to be relative. If the target of
+   * the link is again a link, then that lead is not followed.
+   *
+   * @param link the {@link Path} of the symbolic link.
+   */
+  void makeSymlinkRelative(Path link);
+
+  /**
+   * Symbolic links can point to relative or absolute paths. Here the link is converted to be relative.
+   *
+   * @param link the {@link Path} of the symbolic link.
+   * @param followTarget - {@code true} if the target of the link is again a link, then that lead is followed, until the
+   *        target is not a link. - {@code false} if the target of the link is again a link, then that lead is not
+   *        followed.
+   */
+  void makeSymlinkRelative(Path link, boolean followTarget);
+
+  /**
+   * Creates a symbolic relative link.
+   *
+   * @param source the source {@link Path} to link to.
+   * @param targetLink the {@link Path} where the symbolic link shall be created pointing to {@code source}.
+   */
+  void relativeSymlink(Path targetLink, Path source);
```

Comment:
could be removed if the implementation is deemed unnecessary, see comment in FileAccesImpl.java

Reply:
IMHO it is totally confusing that `symlink` is taking the source as the first argument and the target as second argument and you now created a very similar method where you inverted the semantic of these parameters.
Please also note that my order of args was not chosen by random: symbolic links work this way `ln -s «source» «target»` and that is also intentional as is is analog to `cp «source» «target»`.

My suggestion would be to instead add the method like this and even make relative symlinks the new default
```
void symlink(Path source, Path targetLink, boolean relative);

default symlink(Path source, Path targetLink) {
  symlink(source, targetLink, true);
}
```
In all places where we currently create symlinks (esp. in software folder) we would benefit from relative links.
If we have the need to create an absolute link, we have the ability to provide `false` for the `relative` parameter.

Reply:
Okey, I see your point. I have adjusted it to be like you suggested. Thank you both!